src/ansiblelint/yaml_utils.py

```diff
@@ -874,6 +880,9 @@ def __init__(
         # (see https://stackoverflow.com/a/44314840/1134951)
         # self.Representer.add_representer(
 
+        if version:
+            self.version = version
```

Comment:
Why did you add this? Setting `self.version` actually sets `self._yaml_version` which is already configured above. Why do we need to set it again?